pio-proc/src/lib.rs

```diff
@@ -8,26 +8,36 @@ use syn::parse_macro_input;
 #[proc_macro]
 pub fn pio(item: TokenStream) -> TokenStream {
     let source = parse_macro_input!(item as syn::LitStr);
-    let result = match pio_parser::Program::parse_program(&source.value()) {
-        Ok(p) => {
-            let origin: proc_macro2::TokenStream = format!("{:?}", p.origin()).parse().unwrap();
+    let result = match pio_parser::Parser::<{ pio::RP2040_MAX_PROGRAM_SIZE }>::parse_program(
```

Comment:
i think you should have to pass the size in.
```rust
pio!(pio::RP2040_MAX_PROGRAM_SIZE, "
  ...
");
```

Reply:
Didn't think of that. One challenge is that the parser still needs a constant upper bound for the size of the program, but I think it can be set to 1024 instructions or so for now, and if someone ever refactors the parser to work with `Vec` then that can be used. In the meanwhile the given size is used to determine the size of the `ArrayVec`.

Reply:
This should be fixed now. The program size is limited to 1024 for now, but in the future `Vec` should be used while parsing.